pkg/types/provider.go

```diff
@@ -70,6 +71,34 @@ func (e UnimplementedError) Is(target error) bool {
 	return grpcErrorHasTypeAndMessage(target, string(e), codes.Unimplemented)
 }
 
+type InternalError string
+
+func (e InternalError) Error() string {
+	return string(e)
+}
+
+func (e InternalError) GRPCStatus() *status.Status {
+	return status.New(codes.Internal, e.Error())
+}
+
+func (e InternalError) Is(target error) bool {
+	if e == target {
+		return true
+	}
+
+	return grpcErrorHasTypeAndMessage(target, string(e), codes.Unimplemented)
```

Comment:
An internal error is an unimplemented error? Is that right?

Reply:
No, internal error is for if you have an error that isn't something you expected.  Eg: ChainReader wraps with one if log poller gives it an error.  I think of it like a 5xx, you shouldn't get one.

Unimplemented will be returned when the ChainReader (or codec) isn't initilzide

Reply:
But should this be `codes.Internal` instead of `codes.Unimplemented`?
```suggestion
	return grpcErrorHasTypeAndMessage(target, string(e), codes.Internal)
```